react/components/Toast/Toast.js

```diff
@@ -0,0 +1,169 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import CloseIcon from '../icon/Close'
+
+const DURATION_BASE = 3000
+const DURATION_INCREMENT = 30
+const TRANSITION_DURATION = 160
+
+export default class Toast extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      isOpen: false,
+    }
+  }
+
+  componentDidMount() {
+    if (this.props.visible) {
+      setTimeout(() => {
+        this.open()
+      }, 10)
+      this.startAutoClose()
+    }
+  }
+
+  // Duration increases along with the length of the message
+  getDefaultDuration = () => DURATION_BASE + this.props.message.length * DURATION_INCREMENT
+
+  startAutoClose = () => {
+    this.stopAutoClose()
+    const { duration } = this.props
+
+    if (duration === 0 || duration < 0 || duration === Infinity) {
```

Comment:
Why don't you use less than or equal to `0` (`<= 0`) instead of two different comparisons?

Reply:
@gugabribeiro Good catch—I just added the different conditions one at a time, and had not thought about that.